shared/domains/chelonia/localSelectors.js

```diff
@@ -90,5 +91,23 @@ export default (sbp('sbp/selectors/register', {
         sbp('okTurtles.events/emit', CONTRACTS_MODIFIED_READY, subscriptionSet, { added, removed })
       })
     })
+  },
+  'chelonia/externalStateWait': async function (contractID) {
+    await sbp('chelonia/contract/wait', contractID)
+    const { cheloniaState } = await sbp('chelonia/contract/fullState', contractID)
+    const localState = sbp(this.stateSelector)
+    if (cheloniaState.height >= localState.contracts[contractID]?.height) return
+
+    return new Promise((resolve) => {
+      const removeListener = sbp('okTurtles.events/on', EVENT_HANDLED_READY, (cID) => {
+        if (cID !== contractID) return
+
+        const localState = sbp(this.stateSelector)
+        if (cheloniaState.height >= localState.contracts[contractID]?.height) {
+          resolve()
+          removeListener()
```

Comment:
This seems somewhat odd, I rarely see any code after `resolve()`... would it make sense to flip these two lines so that `removeListener()` is called first?

Reply:
I don't think so. The idea here (and in general, I think) is to resolve when the promise should resolve. The cleanup code comes after `resolve()` because it doesn't affect the settlement of the promise. For example, if `removeListener()` should throw, it makes sense for the promise to still resolve. Investing the order would have a different effect.